### DISCUSSION TITLE
Problems with html_proofer_staging

### DISCUSSION THREAD

**Discussion by MoBosse** [open-sdg/open-sdg#1885] [Q&A]
Hi all,
today a run into a problem when trying to upgrade my production site from a very old version of open sdg to 2.1.0.

First I created a test environment and everything worked perfectly. Then I copied the code into my existing repositories to create our reporting platform. The staging site updated without any issues but when I try to make a PR towards my production-site repo I keep getting errors saying `internally linking to /open-sdg-site-starter/[...], which does not exist`.

It seems to have something to do with the html_proofer_staging. 
I am pretty confused because everything worked with my test environment (staging and production site) and I just copied the files to my original repo...

These are my repositories:
site staging: https://github.com/Nachhaltige-Entwicklung-Deutschland/open-sdg-site-starter
data staging: https://github.com/Nachhaltige-Entwicklung-Deutschland/open-sdg-data-starter

site production: https://github.com/sustainabledevelopment-deutschland/sustainabledevelopment-deutschland.github.io
data production: https://github.com/sustainabledevelopment-deutschland/sdg-data-pub

We need to update the platform to present it at a big event on Thursday, so this is kind of an emergency for me. @brockfanning or anyone else: If you could help us with this, I would be very grateful.

**Comment by sarah-beasley** [comment]
Hi @MoBosse , Did you try to upgrade directly to 2.1.0 from the older version? It might be that you need to upgrade in stages as there are some large changes between many upgrades that could cause errors. For example, if upgrading from around 1.8.0, it would be best to at least upgrade to 2.0.0 first. This may fix the problem. However if it's an even earlier version then you may need to do a couple more in between to capture some more changes. Keep us posted though!

**Reply by sarah-beasley** [reply]
Also @MoBosse , any other errors to do with Ruby - just seen one in your staging site repo, Brock announced a fix for this here which could help - https://github.com/open-sdg/open-sdg/discussions/1884

**Comment by MoBosse** [comment]
Hi @sarah-beasley , thanks for the reply. I may have already solved the problem. I seem to have had a couple of links inside my indicator meta files and in some pages files that the html_proofer for some reason did not like. I deleted all of them for now and the tests stopped failing. I will investigate a bit more about which links caused the fails exactly later. Maybe it will be helpful for other users.
I am still not able to merge the PR but this seem to be another issue.

Thanks about that hint with #1884 but I already implemented that.
